server/service/software.go

```diff
@@ -147,16 +148,21 @@ func getSoftwareEndpoint(ctx context.Context, request interface{}, svc fleet.Ser
 }
 
 func (svc *Service) SoftwareByID(ctx context.Context, id uint, teamID *uint, includeCVEScores bool) (*fleet.Software, error) {
-	if err := svc.authz.Authorize(ctx, &fleet.AuthzSoftwareInventory{TeamID: teamID}, fleet.ActionRead); err != nil {
+	if err := svc.authz.Authorize(ctx, &fleet.Host{TeamID: teamID}, fleet.ActionList); err != nil {
 		return nil, err
 	}
 
 	if teamID != nil {
+		// This auth check ensures we return 403 if the user doesn't have access to the team
```

Comment:
Don't you also have to check if e.g. a team user is allowed to read software in global hosts?

Reply:
If teamID is specified, then this method will not read software in any non-team hosts. It will only read non-team hosts if teamID is not specified, which is the change done by MDM team and already merged into main.

Reply:
Just double checking because I'm a little confused. If `teamID == nil` don't you still have to check:
```go
if err := svc.authz.Authorize(ctx, &fleet.AuthzSoftwareInventory{TeamID: teamID}, fleet.ActionRead); err != nil {
  return nil, err
}
```

Reply:
Ah, I think I see it, ds.SoftwareByID has some filtering depending on the requesting user.

Reply:
When `teamID == nil` SQL code limits the teams to the ones the user can access. The query only checks the user's teams. So, there is no explicit `authz` check for that.